examples/common/pigweed/rpc_services/AttributeAccessor.h

```diff
@@ -0,0 +1,94 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "pw_status/status.h"
+#include <app/AttributeReportBuilder.h>
+#include <app/AttributeValueDecoder.h>
+#include <app/AttributeValueEncoder.h>
+
+/**
+ * Callback class that clusters can implement in order to interpose custom
+ * attribute-handling logic.  An AttributeAccessor instance is associated
+ * with some specific cluster.  A single instance may be used for a specific
+ * endpoint or for all endpoints.
+ *
+ * Instances of AttributeAccessor that are registered via
+ * AttributeAccessorRegistry::Instance().Register will be consulted before taking the
+ * normal attribute access codepath and can use that codepath as a fallback if desired.
+ */
+namespace chip {
+namespace rpc {
+
+class AttributeAccessor
```

Comment:
Suggest rename to `PigweedDebugAccessInterceptor`, so that the registry can be reused by actions service and by attributes service. That way it's not just for Attribute, but we could possibly intercept any of the pigweed RPC in the future, as needed.